src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -272,16 +274,25 @@ private void RegisterActions(CompilationStartAnalysisContext compilationStartCon
                     OperationKind.DynamicMemberReference,
                     OperationKind.DynamicObjectCreation);
 
-                symbolStartContext.RegisterSymbolEndAction(symbolEndContext => OnSymbolEnd(symbolEndContext, hasUnsupportedOperation));
+                // We analyze extension block members as part of the enclosing static class.
+                if (symbolStartContext.Symbol is not INamedTypeSymbol { IsExtension: true })
+                {
+                    symbolStartContext.RegisterSymbolEndAction(symbolEndContext => OnSymbolEnd(symbolEndContext, hasUnsupportedOperation));
+                }
 
                 // Register custom language-specific actions, if any.
                 _analyzer.HandleNamedTypeSymbolStart(symbolStartContext, onSymbolUsageFound);
             }, SymbolKind.NamedType);
 
             bool ShouldAnalyze(SymbolStartAnalysisContext context, INamedTypeSymbol namedType)
             {
+                if (namedType.IsExtension)
+                {
+                    return false;
+                }
```

Comment:
```suggestion
                    return false;
```

Also, consider a comment above the if.  but ok if not present.